docs/hello_nextflow/09_hello_nf-core.md

```diff
@@ -244,4 +244,1050 @@ You have a general idea of what nf-core offers and you know how to run an nf-cor
+
+```console
+nextflow run . -profile docker,test --outdir results -dump-channels
+```
+
+The meta map now has a new key `machineid`, that is empty because we did not specify a value yet:
+
+```console title="Output"
+[DUMP] [['id':'SAMPLE1_PE', 'machineid':[], 'single_end':false], [/nf-core/test-datasets/viralrecon/illumina/amplicon/sample1_R1.fastq.gz, /nf-core/test-datasets/viralrecon/illumina/amplicon/sample1_R2.fastq.gz]]
+[DUMP] [['id':'SAMPLE2_PE', 'machineid':[], 'single_end':false], [/nf-core/test-datasets/viralrecon/illumina/amplicon/sample2_R1.fastq.gz, /nf-core/test-datasets/viralrecon/illumina/amplicon/sample2_R2.fastq.gz]]
+[DUMP] [['id':'SAMPLE3_SE', 'machineid':[], 'single_end':true], [/nf-core/test-datasets/viralrecon/illumina/amplicon/sample1_R1.fastq.gz, /nf-core/test-datasets/viralrecon/illumina/amplicon/sample2_R1.fastq.gz]]
+```
+
+We have also prepared a new samplesheet, that has the `machineid` column. You can overwrite the existing input with this command:
+
+```console
+nextflow run . -profile docker,test --outdir results -dump-channels --input ../data/machineid_samplesheet.csv 
+```
+
+This populates the `machineid` and we could access it in the pipeline:
+
+```console
+[DUMP] [['id':'SAMPLE1_PE', 'machineid':'myfavorite_machine', 'single_end':false], [/nf-core/test-datasets/viralrecon/illumina/amplicon/sample1_R1.fastq.gz, /nf-core/test-datasets/viralrecon/illumina/amplicon/sample1_R2.fastq.gz]]
+[DUMP] [['id':'SAMPLE2_PE', 'machineid':'worst_machine', 'single_end':false], [/nf-core/test-datasets/viralrecon/illumina/amplicon/sample2_R1.fastq.gz, /nf-core/test-datasets/viralrecon/illumina/amplicon/sample2_R2.fastq.gz]]
+[DUMP] [['id':'SAMPLE3_SE', 'machineid':'myfavorite_machine', 'single_end':true], [/nf-core/test-datasets/viralrecon/illumina/amplicon/sample1_R1.fastq.gz, /nf-core/test-datasets/viralrecon/illumina/amplicon/sample2_R1.fastq.gz]]
+```
+
+### Use the new meta key in the pipeline
+
+<!-- TODO No good idea yet on what to do here -->
+
+### Takeaway
+
+You know how to adapt the samplesheet to add new meta information to your files. 
+
+### What's next?
+
+In the next step we will add a module that is not yet in nf-core.
+
+---
+
+## Create a custom module for your pipeline
+
+nf-core offers a comprehensive set of modules that have been created and curated by the community. However, as a developer, you may be interested in bespoke pieces of software that are not apart of the nf-core repository or customizing a module that already exists.
+
+In this instance, we will write a local module for the QC Tool [FastQE](https://fastqe.com/), which computes stats for FASTQ files and print those stats as emoji.
+
+This section should feel familiar to the `hello_modules` section.
+
+### Create the module
+
+Start by using the nf-core tooling to create a sceleton local module. It will prompt you to type in the tool name `fastqe`, for the remaining fields press `enter` to accpet the default: 
```

Comment:
```suggestion
Start by using the nf-core tooling to create a kceleton local module. It will prompt you to type in the tool name `fastqe`, for the remaining fields press `enter` to accept the default: 
```

Reply:
Also I'd love it if we discouraged local modules and encouraged contribution to nf-core from the get-go for the benefits on testing, community input etc

Reply:
yeah true. I moved the note up now as the first thing.